### PR TITLE
ICU-20243 Add YML file for Exhaustive Tests in the CI builds.

### PR DESCRIPTION
**Pull request by jefgen**
This adds a separate YML file for the Exhaustive Tests configuration so that the Azure Pipelines build can be run separately from the regular PR builds.

Once this is added, we can decided if we want to run this on every single PR, or just on the master branch on scheduled builds.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20243
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by jefgen** [comment]
(Assigning to @sffc since he was interested in the Exhaustive tests the other day.)

File: .ci-builds/.azure-exhaustive-tests.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Note: The exhaustive tests for J take longer than the C tests.
+- job: ICU4J_OpenJDK_Ubuntu_1604
+  displayName: 'J: Linux OpenJDK (Ubuntu 16.04)'
+  timeoutInMinutes: 100
+  pool:
+    vmImage: 'Ubuntu 16.04'
+    demands: ant
+  steps:
+    - script: |
+        echo "Building ICU4J" && cd icu4j && ant init && ant check
```
Comment (sffc) [inline .ci-builds/.azure-exhaustive-tests.yml]: Why not dive straight into exhaustive?

File: .ci-builds/.azure-exhaustive-tests.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      displayName: 'Build and Test'
+      env:
+        BUILD: ICU4J
+    - script: |
+        cat `find out/junit-results -name "*.txt" -exec grep -l FAILED {} \;`
+      condition: failed() # only run if the build fails.
+      displayName: 'List failures (if any)'
```
Comment (sffc) [inline .ci-builds/.azure-exhaustive-tests.yml]: List failures on exhaustive failures?

Comment (jefgen) [reply .ci-builds/.azure-exhaustive-tests.yml]: Ah, good point, will update.

File: .ci-builds/.azure-exhaustive-tests.yml (hunk 1 above)
Comment (jefgen) [reply .ci-builds/.azure-exhaustive-tests.yml]: I suppose we could. Thanks.

**Comment by jefgen** [comment]
> Will this run always, or only on master?

So right now, this won't run anywhere at all.
Once the file is added to the repo then we can tell Azure Pipelines to build (using this config file) on triggers, such as:
- Every PR.
- Commits to master.
- A fixed schedule (ie: set time).

We can start with perhaps just building on master (or a set time) and change it later as desired.

**Comment by jefgen** [comment]
Note: The Azure builds are showing as "Queued" despite successfully completing.

**Pull request merged**